src/VisualStudio/TestUtilities/IntegrationHelper.cs

```diff
@@ -0,0 +1,209 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.IO;
+using System.Net;
+using System.Security;
+using System.Text;
+using System.Threading;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.VisualStudio.Test.Utilities.Interop;
+
+namespace Roslyn.VisualStudio.Test.Utilities
+{
+    internal static class IntegrationHelper
+    {
+        public const int DefaultTimeout = 5000; // 5 Seconds
+        public const int DefaultSleepDelay = 100; // 1/10th of a Second
+
+        private static Stopwatch _stopwatch = Stopwatch.StartNew();
+
+        public static long CurrentTickCount
+        {
+            get
+            {
+                return _stopwatch.ElapsedMilliseconds;
+            }
+        }
+
+        public static void AbortThread(Thread thread, int timeout = DefaultTimeout)
+        {
+            if ((thread != null) && (!thread.Join(timeout)))
+            {
+                try
+                {
+                    thread.Abort();
```

Comment:
Because killing `xUnit` is undesirable and we may be interacting with another process (VS). We attempt to stop the thread normally and then kill it if everything else fails. 

In the current setup, the thread cleanup is done for the Modal and Timeout watcher threads. We attempt to let them 'die' gracefully and abort them if failure. The remainder of the cleanup needs to happen even if the threads fail to abort.